src/Chainweb/Pact/Backend/InMemoryCheckpointer.hs

```diff
@@ -53,21 +59,34 @@ initInMemoryCheckpointEnv loggers logger gasEnv = do
                     (doSave inmem)
                     (doDiscard inmem)
                     (doGetLatest inmem)
-                    id  -- in-mem doesn't require tx rewind
+                    noop        -- in-mem doesn't batch
+                    noop
+                    (doDiscard inmem)
                     (doLookupBlock inmem)
                     (doGetBlockParent inmem)
+                    (doRegisterSuccessful inmem)
+                    (doLookupSuccessful inmem)
             , _cpeLogger = logger
             , _cpeGasEnv = gasEnv
             })
+  where
+    noop = return ()
 
 doRestore :: DbEnv PureDb -> MVar Store ->  Maybe (BlockHeight, ParentHash) -> IO PactDbEnv'
 doRestore _ lock (Just (height, hash)) =
     modifyMVarMasked lock $ \store -> do
-      let bh = (pred height, hash)
+      let ph = pred height
+      let bh = (ph, hash)
+      let filt (_, mp) = (mempty, HMS.filterWithKey (\_ (v, _) -> v <= ph) mp)
       case HMS.lookup bh (_theStore store) of
             Just dbenv -> do
               mdbenv <- newMVar dbenv
-              let !store' = store { _lastBlock = Just bh, _dbenv = mdbenv }
+              played <- readMVar (_playedTxs store) >>= newMVar . filt
```

Comment:
Is the result of applying `filt` forced, including the hash set in the second tuple field?

Reply:
We don't care -- the in memory checkpointer is not in use. I'll fix it anyways